crates/subspace-transaction-pool/src/lib.rs

```diff
@@ -218,19 +270,55 @@ where
                                 }
                             }
                         }
-                        .boxed();
+                            .boxed();
                     }
                 }
             }
             Err(err) => {
                 return async move {
-                        Err(sc_transaction_pool::error::Error::Blockchain(err.into()))
-                    }
+                    Err(sc_transaction_pool::error::Error::Blockchain(err.into()))
+                }
                     .boxed();
             }
         }
 
-        self.inner.validate_transaction(at, source, uxt)
+        chain_api.validate_transaction(at, source, uxt)
```

Comment:
Can we extend the existing `PreValidationObject` concept to a more general `PreValidateTransaction` to make `subspace-transaction-pool` modular and composable? The new `ValidateExtrinsic` abstraction is made for an opportunity to perform some pre-validation for the incoming transactions, which is no different from what we have done for verifying the fraud proof and receipts.

So we have different pre-validation requirements for different chains:

- Primary chain: validate the fraud proof from the system domain and the bundles from all domains.
- System domain: validate the fraud proof from the non-system domain.
- Core domain: validate the xdm.

`FraudProofVerifier` already exists, `BundleValidator` is introduced in #1145, now we just need to another `XdmValidator`. If we have a trait defined `PreValidateTransaction::pre_validate_transaction` (just to be clear this happens before the regular `validate_transaction`), then we can build the transaction pool using the same interface in a composble way,

- Primary chain: `PrimaryChainPreValidator` (`FraudProofVerifier` + `BundleValidator`)
- System domain: `SystemDomainPreValidator` (`FraudProofVerifier`)
- Core domain: `CoreDomainPreValidator` (`XdmValidator`)

Reply:
Yes we can. As mentioned above, main focus of this PR was to integrate the client side changes to verify xdm messages. So I have abstracted code as required. I did not touch that Fraud proof verifier even though I think it should not be part of transaction-pool pallet. But with this change, it is easier to combine multiple Validators.

> FraudProofVerifier already exists, BundleValidator is introduced in https://github.com/subspace/subspace/pull/1145, now we just need to another XdmValidator.

This changes ideally make transaction pool crate abstract over Validators(s). 

As for finding keeping xdm_validator in domain-executor crate is beacuse i couldn't find a better crate. Though i believe another sc-crate just for messages might not be required IMO

Reply:
Another reason, PreValidation was used as well due to importing `domains/*` to `crates/*`. I think it would be valid to to create another crate that could be used for Prevalidation but incrementally.

Reply:
>Another reason, PreValidation was used as well due to importing domains/* to crates/*

I don't follow.

Although I feel we do have a chance to architect it better, it's fine for me to take it for now and refactor it later (this may also avoid some unnecessary early optimization).

Reply:
> (this may also avoid some unnecessary early optimization).

This. I was trying to say this 🤔